src/__Private/Visitor/FragmentInfo.hack

```diff
@@ -0,0 +1,59 @@
+namespace Slack\GraphQL\__Private;
+
+use namespace HH\Lib\C;
+use namespace Graphpinator\Parser;
+use type \Slack\GraphQL\__Private\Utils\Stack;
+
+final class FragmentInfo extends ASTVisitor {
```

Comment:
Couple thoughts @mwildehahn and @jjergus :

1. What do ya'll think of consolidating this and `VariableInfo` into one class, `DependencyTracker` or something like that?
2. Thought on throwing an error if someone calls `getFragments` (or any of these methods) before the AST has been fully traversed? Anything returned from these methods will be inaccurate till then.